tests/components/songpal/test_init.py

```diff
@@ -0,0 +1,66 @@
+"""Tests songpal setup."""
+from homeassistant.components import songpal
+from homeassistant.setup import async_setup_component
+
+from . import (
+    CONF_DATA,
+    _create_mocked_device,
+    _patch_config_flow_device,
+    _patch_media_player_device,
+)
+
+from tests.async_mock import patch
+from tests.common import MockConfigEntry
+
+
+def _patch_media_setup():
+    """Patch media_player.async_setup_entry."""
+
+    async def _async_return():
+        return True
+
+    return patch(
+        "homeassistant.components.songpal.media_player.async_setup_entry",
+        side_effect=_async_return,
+    )
+
+
+async def test_setup_empty(hass):
+    """Test setup without any configuration."""
+    with _patch_media_setup() as setup:
+        assert await async_setup_component(hass, songpal.DOMAIN, {}) is True
+        await hass.async_block_till_done()
+    setup.assert_not_called()
+
+
+async def test_setup(hass):
+    """Test setup the platform."""
+    mocked_device = _create_mocked_device()
+
+    with _patch_config_flow_device(mocked_device), _patch_media_setup() as setup:
+        assert (
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5518609)